public/visualization_fn.ts

```diff
@@ -0,0 +1,168 @@
+/*
```

Comment:
could you move this file to `data_load` sub folder?

Reply:
Moved the file as requested